src/sparse/prod.rs

```diff
@@ -94,10 +95,11 @@ pub fn mul_acc_mat_vec_csr<N, I, Iptr, V>(
     }
 
     for (row_ind, vec) in mat.outer_iterator().enumerate() {
+        let tv = res_vec.get_mut(row_ind).unwrap(); // this unwrap is ok because we did the check
```

Comment:
I'm not sure this comment is clear, I think the index is in bounds because `CsMat`guarantees its indices to be in bounds, and because the dimensions have been checked to be in agreement.

It's however a very good thing to move this indexing out of the inner loop, this means the index checking will not be too much of a cost, good catch!

Reply:
I added some more comment to explain why I think it's ok to `unwrap` there. My understanding is that the earlier check `mat.rows() != res_vec.len()` in that function ensures the `unwrap` is fine.

Reply:
The new comment is clearer, thanks!